pkg/providers/aws/provider_redissnapshot_test.go

```diff
@@ -34,7 +34,9 @@ type elasticacheClientMock struct {
 	DescribeReplicationGroupsFunc func(in1 *elasticache.DescribeReplicationGroupsInput) (*elasticache.DescribeReplicationGroupsOutput, error)
 	CreateSnapshotFunc            func(in1 *elasticache.CreateSnapshotInput) (*elasticache.CreateSnapshotOutput, error)
 	DeleteSnapshotFunc            func(in1 *elasticache.DeleteSnapshotInput) (*elasticache.DeleteSnapshotOutput, error)
-	calls                         struct {
+	DeleteCacheSubnetGroupFn      func(in1 *elasticache.DeleteCacheSubnetGroupInput) (*elasticache.DeleteCacheSubnetGroupOutput, error)
```

Comment:
I know you called this out, but I think we should only have one elasticacheMock struct, i think it would be valuable to refactor this struct into the other existing struct.

Reply:
This was a bit of a mess that I created so If you want to create a separate ticket for it I would be happy to do the refactor 🙂

Reply:
This was a bit of a mess that I created so If you want to create a separate ticket for it I would be happy to do the refactor 🙂